test/jdk/java/nio/channels/Channels/ClosedNetworkChannels.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 8329190
+ * @summary Test that I/O operations on a closed network channel throw ClosedChannelException
+ *    and not AsynchronousCloseException
+ * @run junit ClosedNetworkChannels
+ */
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousCloseException;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.DatagramChannel;
+import java.nio.channels.NotYetConnectedException;
+import java.nio.channels.Pipe;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+class ClosedNetworkChannels {
+
+    /**
+     * An operation that does not return a result but may throw an exception.
+     */
+    @FunctionalInterface
+    interface ThrowingRunnable {
+        void run() throws Exception;
+    }
+
+    /**
+     * Assert that the given operation throws ClosedChannelException.
+     */
+    private void assertThrowsCCE(ThrowingRunnable op) throws Exception {
+        try {
+            op.run();
+            fail();
+        } catch (AsynchronousCloseException e) {
+            fail(e + " thrown");
```

Comment:
I think for debugging any unexpected failures, it might be better to include the original exception too in the failure. Like:
```java
fail(e + " thrown", e);
```

Reply:
yes, that is a reasonable suggestion. I will, if you don't mind, leave things as is, for the simple reason, I would need to submit such a change on repeat mode through MACH5, as it's for unexpected exceptions, and this can take 5/6 days for the job to get through MACH5, depending on resource availability, and I have gone through that cycle three times already for this fix

Reply:
Leaving it in the current form is OK with me. We can revisit that line if/when we modify this test later.